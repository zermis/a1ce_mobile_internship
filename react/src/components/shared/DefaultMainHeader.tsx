import React from 'react';
import { Text, StyleSheet, Platform, Dimensions } from 'react-native';

interface Props {
    children?: string;
    style?: object;
}

const DefaultMainHeader = (props: Props) => {
    return (
        <Text style={{ ...styles.title, ...props.style }}>
            {props.children}
        </Text>
    );
};

const styles = StyleSheet.create({
    title: {
        color: '#fff',
        fontSize: Dimensions.get('window').height * 0.045,
        fontWeight: Platform.OS === 'ios' ? '600' : '700',
        letterSpacing: -0.5
    }
});

export default DefaultMainHeader;
