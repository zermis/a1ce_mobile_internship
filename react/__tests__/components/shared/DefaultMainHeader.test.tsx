import DefaultMainHeader from '../../../src/components/shared/DefaultMainHeader';
import renderer from 'react-test-renderer';

it('render default main header correctly', function () {
    var tree = renderer
        .create(<DefaultMainHeader children="Test" style={{}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
