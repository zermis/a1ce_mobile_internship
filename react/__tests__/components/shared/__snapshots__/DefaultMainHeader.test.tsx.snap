// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default main header correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontSize": 60.03,
      "fontWeight": "600",
      "letterSpacing": -0.5,
    }
  }
>
  Test
</Text>
`;
